crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
I agree that there's no direct way to indicate an error, but the in |ptr| parameter is always NULL, under the assumption you only call CRYPTO_new_ex_data() once on an object (not a good idea to call more than once!).

Indirectly, you can look at the result of CRYPTO_get_ex_data() to see if the new callback succeeded (this of course, assumes you know the new callback sets the pointer via CRYPTO_set_ex_data()).